defect #2451038 Octane cannot correctly open test run report

andreibangau99 opened the pull request:
- Created logic so that the test report URL is correct when running tests on different machines inside the same Job

Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!

- [ ] Proper pull request title - concise and clear for others.
- [ ] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [ ] Proper Jira ticket - Number, Link in pull request description.
- [ ] The PR can is merged on your machine without any conflicts.
- [ ] The PR can is built on your machine without any (new) warnings.
- [ ] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [ ] Add unit tests with new features.
- [ ] If you added any dependency to the POM - Please update grount

Commit:
defect #2451038 Octane cannot correctly open test run report

- Created logic so that the test report URL is correct when running tests on different machines inside the same Job

nissimshitrit commented on src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitExtension.java:
```diff
@@ -230,22 +232,30 @@ public GetJUnitTestResults(Run<?, ?> build, HPRunnerType hpRunnerType, List<File
+				List<Node> nodes = new ArrayList<>();
+
+				allWorkspaces.forEach(workspace-> nodes.add(JenkinsUtils.getCurrentNode(workspace)));
+				nodes.forEach(node -> this.nodeNames.add(node != null && !node.getNodeName().isEmpty() ? node.getNodeName() : ""));
```
It seems that there is case when node name is empty as far as I remember it a case when UFT is run on Jenkins machine itself and not on one of it's agents, can you please try to check if your code covers case the UFT test run with not agents?
Secondly instead of define array and fill it up there more convenient way to do it in Java like this
allWorkspaces.stream().filter(workspace-> nodes.add(JenkinsUtils.getCurrentNode(workspace)).map(workspace-> JenkinsUtils.getCurrentNode(workspace))).collect(collectors.toList())

andreibangau99 replied:
Indeed, when running on the Jenkins machine itself, the node name remains empty. I tested the behavior, and the reports URLs are created correctly in this case, with the path to the 'artifacts' folder from the Jenkins machine.

For the second part, I changed the code accordingly. Please let me know if I missed something.

nissimshitrit commented on src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitXmlIterator.java:
```diff
@@ -261,24 +261,34 @@ private void handleJUnitTest(XMLEvent event) throws XMLStreamException, IOExcept
-                            cleanedTestName = optional.get();
-                            createdTests.remove(cleanedTestName);
+                            String nodeTestString = optional.get();
+                            String node = nodeTestString.split("/")[0];
```
can you please add check before split that you have "/" exist to prevent unplanned errors

andreibangau99 replied:
Added the check.

Commit:
defect #2451038 Octane cannot correctly open test run report

- Created logic so that the test report URL is correct when running tests on different machines inside the same Job
+added additional checks and reformats
